### DISCUSSION TITLE
Bar width

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#1052] [Q&A] [answered]
Is there any way to control the bar width on bar chat type plots? And also control the spacing/padding between bars?

In the chart below (not rendered with Observable) the bars are too thick for the stacked bar chart. I'd like to be able to control the bar width and also the spacing between them (i.e. not have huge gaps between bars).

Do we have to let the chart have auto width to do this? And is there a default min/max bar width.

![stacked-bar-chart](https://user-images.githubusercontent.com/1482075/190900582-2050ddbe-3b15-443f-baaf-bed0adc304ef.png)

**Comment by Fil** [comment, marked as answer]
Yes, you can fine tune the band scales with the settings described in https://github.com/observablehq/plot#position-options : padding, align, paddingInner, paddingOuter — on top of width, marginLeft, marginRight, and inset.

**Reply by dgwyer** [reply]
Those are great options to have but I can't seem to get the same result as in my original post above with just the bar width reduced (i.e. same padding between stacked bars).

I can reduce the bar size by specifying values for inset/padding but the spacing remains between the bars. https://observablehq.com/@dgwyer/untitled

![image](https://user-images.githubusercontent.com/1482075/191222514-c9ec7366-0e64-4f73-91a0-13af8ae688f5.png)

I guess the reason is that there's a minimum chart width to layout all bars inside?

**Reply by Fil** [reply]
there is a top-level _width_ option, too — it defaults to 640.
